speed-transition.lua

```diff
@@ -63,34 +84,47 @@ end
 
 function skipval()
    local skipval = mp.get_property_native("demuxer-cache-duration", 5)
-   skipval = clamp(skipval, 1, 5)
    if nextsub > 0 then
-      if nextsub-skipval-leadin <= 0 or directskip then
-         skipval = nextsub - leadin
+      if directskip then
+         skipval =  nextsub - leadin
+      elseif nextsub-skipval-leadin <= 0 then
+         skipval =  clamp(nextsub-leadin, 0, maxskip)
       end
+   elseif directskip then
+      skipval = clamp(skipval-leadin, 1, nil)
    else
-      skipval = clamp(skipval-leadin, 0, nil)
-      if skipval == 0 then
-         skipval = clamp(skipval-1, 1, nil)
-      end
+      skipval = clamp(skipval-leadin, 1, maxskip)
    end
    return skipval
 end
 
 function speed_transition(_, sub)
    if state == 0 then
-      if sub == "" then
+      if sub == "" or (sub and shouldIgnore(sub)) then
          last_speedup_zone_begin = speedup_zone_begin
          nextsub, shouldspeedup, speedup_zone_begin = check_should_speedup()
          mark = speedup_zone_begin
          speedup_zone_end = mark+nextsub
-         if shouldspeedup then
+         if shouldspeedup or (skipmode and not firstskip) then
```

Comment:
I hoped for a less confusing way to write this but it's effective in making the first skip not have to wait for delay.